app/common/auth/__init__.py

```diff
@@ -30,6 +30,8 @@ def request_a_link_to_sign_in() -> ResponseReturnValue:
         email = cast(str, form.email_address.data)
 
         user = get_or_create_user(email_address=email)
+        if user is None:
+            abort(404)
```

Comment:
Is this possible with the default behaviour of `get_or_create_users`? It would always create a user; how would we get back None? I think you're addressing a type hinting issue here that doesn't reflect the reality of the code execution path - let's fix the problem, not the symptom. :)